### ISSUE TITLE
Running multiple fakereplace instances seems to cause some issues

### ISSUE THREAD

**Issue by fbricon** [fakereplace/fakereplace#21]
So, here's a fun use case:

I have my eclipse instance, where I develop some Eclipse plugins. I test them by running an hosted, 2nd eclipse instance. I want to leverage fakereplace in my Eclipse plugin development workflow, so the 2nd instance is started with this: 
```
-Xbootclasspath/a:"/Users/fbricon/Dev/projects/wildfly-hive/org.jboss.tools.servers.wildfly.swarm.core/extras/fakereplace-dist-1.0.0.Alpha4-SNAPSHOT.jar" -javaagent:"/Users/fbricon/Dev/projects/wildfly-hive/org.jboss.tools.servers.wildfly.swarm.core/extras/fakereplace-dist-1.0.0.Alpha4-SNAPSHOT.jar=packages=org.jboss.tools" -Dorg.osgi.framework.bootdelegation=org.fakereplace.*
```
The goal is for the 2nd instance plugins to pick up code changes from the host. So it starts correctly, I can see in the eclipse log that:
```
Fakereplace listening on port 6555
```

Then in the hosted instance, I start a wildfly swarm process (via the plugin I'm developing) with

```
-Xbootclasspath/a:"/Users/fbricon/Dev/projects/wildfly-hive/org.jboss.tools.servers.wildfly.swarm.core/extras/fakereplace-dist-1.0.0.Alpha4-SNAPSHOT.jar" -javaagent:"/Users/fbricon/Dev/projects/wildfly-hive/org.jboss.tools.servers.wildfly.swarm.core/extras/fakereplace-dist-1.0.0.Alpha4-SNAPSHOT.jar=packages=com.redhat.developers.msa.hola,port=6557"
```

Now, *every* time the swarm process starts, my host eclipse instance logs this:
```
java.io.EOFException
	at java.io.DataInputStream.readInt(DataInputStream.java:392)
	at org.fakereplace.server.FakereplaceProtocol.run(FakereplaceProtocol.java:79)
	at org.fakereplace.server.FakereplaceServer.run(FakereplaceServer.java:44)
	at java.lang.Thread.run(Thread.java:745)
```

This doesn't happen when the hosted eclipse is started without fakereplace.

Anyways, the freaking cool thing is Eclipse still manages to handle method refactoring that would normally require a JVM restart. And that's awesome :-) (cc @maxandersen)

**Comment by stuartwdouglas** [comment]
Odd, that looks like something is attempting to connect to the remote port.

I might just stop fakereplace listening unless the port argument is specified, as it is only used for the remote maven plugin at the moment.

**Comment by fbricon** [comment]
Sounds like a good thing to do anyways. And if you could cut an Alpha4 release, I guess I'd have no other choice than to blog about it ;-)

**Comment by stuartwdouglas** [comment]
I have made it no longer on by default, and the 'port' argument is now 'server', specifying it with no argument will make it listen on the default port.

**Comment by fbricon** [comment]
That was fast :-) 
But is it possible to display something to ensure fakereplace is started?

**Comment by stuartwdouglas** [comment]
Done

**Comment by stuartwdouglas** [comment]
I have cut the release, although I think it can take a few hours to sync to central

**Comment by fbricon** [comment]
So the new release is available, I no longer see the exception in the log, I see fakereplace is running on my hosted eclipse, but now, hot code replacement doesn't work at all on the eclipse code O_o. It's late so I probably fucked up something but I swear it used to work at some point. Will investigate tomorrow and open a new ticket if needed.

**Comment by stuartwdouglas** [comment]
How were you doing the replacement? Are you using JDWP, or are you using the Fakereplace protocol?

**Comment by stuartwdouglas** [comment]
Did you figure out what the issue is?

**Comment by fbricon** [comment]
Nope. I tried to rebuild fakereplace from a previous commit but, can't make it work. I'm really puzzled.

I know running that kind of agent with Eclipse is not trivial due to some OSGi niceties. Last time I tried with Spring loaded, it didn't work either https://github.com/spring-projects/spring-loaded/issues/114 (but I should retry that too)
The clever folks at zeroturnaround managed to get it working for JRebel though.
